airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -159,6 +159,12 @@ def start_java_dataflow(self, task_id, variables, dataflow):
         self._start_dataflow(
             task_id, variables, dataflow, name, ["java", "-jar"])
 
+    def start_template_dataflow(self, task_id, variables, parameters, environment,
```

Comment:
I wonder if it makes sense to include execution_date in here as well. IIRC, the task_id is just the name of the task. I think you're trying to uniquely identify task executions by appending a UUID to the end of the task_id name. While this solves the problem, it doesn't make it too easy to differentiate which task execution corresponds to which UUID. Perhaps something like task_id + execution_date + UUID? It might even be worth including the retry number of the task as well, though I don't think that's included in the operator context right now.

Reply:
Good point @criccomini , will add that execution_date once you are happy with the previous changes.

Thanks.